### ISSUE TITLE
Recipe Error - REPLACE Rule After JITTER Does Not Replace

### ISSUE THREAD

**Issue by wetzelj** [pydicom/deid#173]
When testing the ability to use variable fields in recipes, we discovered that using the REPLACE keyword in a recipe for values did not work.  Both samples below should yield the same results to the file header.

**Failed:**
```
JITTER endsWith:Date -5814
REPLACE StudyDate var:myOtherDate
```
**Successful:**
```
JITTER endsWith:Date -5814
REMOVE StudyDate
ADD StudyDate var:myOtherDate
```
 
The real issue here:  When a REPLACE rule is executed following a previous JITTER rule, the replace is not executed.  When a REPLACE rule is executed independently, it does function appropriately.  

Failing unit tests have been committed to the [wetzelj/deid INFOR-646](https://github.com/wetzelj/deid/tree/fix/INFOR-646) branch to demonstrate this issue.  I have not yet done investigation of cause/fix.

**Comment by vsoch** [comment]
@wetzelj I can take a look this weekend if you don't get to it.

**Comment by wetzelj** [comment]
@vsoch - I looked into this a little bit today and I think I'm going to need your help on this one whenever you have a few moments.  I can see what's going on, but I don't understand it.

In a successful REPLACE action (I was using test_replace_with_constant to see this), when the value is replaced in parser.py - add_field() - on [line 397](https://github.com/pydicom/deid/blob/5f03d9f097ad86524c7afb28fbdc10cf1961d62c/deid/dicom/parser.py#L397), the statement updates the value of the item in self.fields as well as self.dicom.

Before executing line 397:
![image](https://user-images.githubusercontent.com/32198293/123868956-62741680-d8fe-11eb-8bcb-7934f099f4d4.png)
After executing line 397:
![image](https://user-images.githubusercontent.com/32198293/123869029-7881d700-d8fe-11eb-96c4-0ed2d5d80ea4.png)


In the error scenario (using test_jitter_replace_compounding), when the value is replaced in parser.py, this same line of code updates the value in self.fields, but fails to update self.dicom.
Before executing line 397:
![image](https://user-images.githubusercontent.com/32198293/123869367-e0d0b880-d8fe-11eb-874f-70122fa88b8d.png)
After executing line 397:
![image](https://user-images.githubusercontent.com/32198293/123869429-f8a83c80-d8fe-11eb-9699-2ed3505f7464.png)

Later on, when we perform the pydicom.save_dicom() action in header.py ([line 154](https://github.com/pydicom/deid/blob/5f03d9f097ad86524c7afb28fbdc10cf1961d62c/deid/dicom/header.py#L155)), we're saving parser.dicom.  Since the value in parser.dicom was never updated, the header is saved with the old value.

**Comment by vsoch** [comment]
I think that's the issue - when we run jitter_timestamp instead of just saving the value to self.dicom, we also need to update self.fields (which isn't done).

**Comment by vsoch** [comment]
okay I have a fix! I'm going to have dinner but I'll PR after.

**Comment by vsoch** [comment]
Fixed by #178 .

@wetzelj time for a release?

**Comment by wetzelj** [comment]
Yep!  I think we're good to go.

**Comment by wetzelj** [comment]
Fixed with 0.2.25
